twtxt/cache.py

```diff
@@ -0,0 +1,66 @@
+"""
+    twtxt.cache
+    ~~~~~~~~~~~
+
+    This module implements a caching system for storing Tweets.
+
+    :copyright: (c) 2016 by buckket.
+    :license: MIT, see LICENSE for more details.
+"""
+
+import os
+import shelve
+
+import click
+
+
+class Cache:
+    cache_dir = click.get_app_dir("twtxt")
+    cache_name = "cache"
+
+    def __init__(self, cache_file, cache):
+        self.cache_file = cache_file
+        self.cache = cache
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cache.close()
+
+    @classmethod
+    def from_file(cls, file):
+        """Try loading given cache file."""
+        try:
+            cache = shelve.open(file)
+            return cls(file, cache)
+        except:
+            raise ValueError("Cache file is invalid or can’t be accessed.")
+
+    @classmethod
+    def discover(cls):
+        """Make a guess about the cache file location an try loading it."""
+        file = os.path.join(Cache.cache_dir, Cache.cache_name)
+        return cls.from_file(file)
+
+    def is_cached(self, url):
+        return True if url in self.cache else False
+
+    def last_modified(self, url):
+        return self.cache[url]["last_modified"]
```

Comment:
I thought about catching the exception for KeyError here .. and return empty list like before.. Makes it more robust. I would expect a None value (or similar) for last_modified instead of an exception if it never was cached.